plugins/graphiql/dev/index.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { FC } from 'react';
+import ReactDOM from 'react-dom';
+import { BrowserRouter } from 'react-router-dom';
+import HomeIcon from '@material-ui/icons/Home';
+import { ThemeProvider, CssBaseline } from '@material-ui/core';
+import {
+  createApp,
+  SidebarPage,
+  Sidebar,
+  SidebarItem,
+  SidebarSpacer,
+  ApiRegistry,
+} from '@backstage/core';
+import { lightTheme } from '@backstage/theme';
+import { plugin, GraphQLEndpoints, graphQlBrowseApiRef } from '../src';
+
+const graphQlBrowseApi = GraphQLEndpoints.from([
+  GraphQLEndpoints.create({
+    id: 'gitlab',
+    title: 'GitLab',
+    url: 'https://gitlab.com/api/graphql',
+  }),
+  GraphQLEndpoints.create({
+    id: 'countries',
+    title: 'Countries',
+    url: 'https://countries.trevorblades.com/',
+  }),
+]);
+
+const app = createApp();
+app.registerApis(ApiRegistry.from([[graphQlBrowseApiRef, graphQlBrowseApi]]));
+app.registerPlugin(plugin);
```

Comment:
Yeah idd, I'd like to not use it for basic stuff like URLs, titles, whatnot. Also because it's things we potentially want to share with the backend.

In this case it's actually not always static config though. Once we got GitHub auth set up I plan to add a GitHub GraphQL tab, with automatic auth.

Reply:
That would generally be a custom fetch that adds an Auth header right?

Reply:
Still sounds like a candidate for wiring up in App.tsx so to speak.